nova/src/circuits/supernova/mod.rs

```diff
@@ -501,33 +528,44 @@ pub(crate) mod tests {
     pub struct TestCircuit<F: Field>(PhantomData<F>);
 
     impl<F: PrimeField> NonUniformCircuit<F> for TestCircuit<F> {
```

Comment:
this looks good! i think having the circuit compute `pc_next` without being given `pc` directly will be more natural in the actual VM context than it is here. since there `z` already is a commitment to a whole bunch of values (memory etc), including the risc-v pc. i'm imagining that we'd have something like `pc = 0` for the VM circuit, and precompiles would probably just always return `0` for `compute_selector()` and the VM circuit would derive the new `pc` from the program stored in memory.